### ISSUE TITLE
Error if node signal stride does not match dtype size

### ISSUE THREAD

**Issue by hunse** [nengo/nengo#1184]
The following code fails
```python
import numpy as np
import nengo

y = np.fft.ifft(np.arange(5)).real

with nengo.Network() as model:
    u = nengo.Node(y)
    up = nengo.Probe(u)

with nengo.Simulator(model):
    pass
```
with
```
  File "/home/ehunsber/workspace/nengo/nengo/builder/signal.py", line 254, in init
    dtype=x.dtype, buffer=self[signal.base].data)
ValueError: strides is incompatible with shape of requested array and size of buffer
```
This is because `y` has a stride of 16 (since it's a view into a complex data type), but the data type size is 8. This causes a problem somewhere.

This is easy to work around in this case (just copy `y` so it's contiguous, e.g. `y = np.array(y)`), but the error message is kind of cryptic.

**Comment by jgosmann** [comment]
This seems like it might be a duplicate of #1156 which should be fixed by #1157.

**Comment by jgosmann** [comment]
Unfortunately, #1157 does not fix this particular problem, but this might still be related to #1156.

**Comment by hunse** [comment]
Hmm, I'm surprised. It definitely seems related to #1156. Can we just check if inputs are C contiguous, and if not make a C contiguous copy?

**Comment by jgosmann** [comment]
I've got a rough idea what's going on now. I have to go to a seminar in a few minutes, but will explain later.

**Comment by jgosmann** [comment]
1. An array `x` with 16-byte items is created.
2. A view `y` of `x` using only 8-byte of each item is created and used as node output.
3. The signal `s` is constructed from this view `y` inheriting the strides of 16 bytes (which is correct with respect to the base of the `initial_value`)
4. A signal view `v` is constructed from `s` which also inherits the 16 byte strides.
5. When the signals are initialized (i.e. the memory is allocated), `s` will be initialized by copying `y` (at least in this case, there might be cases where a view is used) which produces a new array `z` with 8-byte elements and a stride of 8.
6. The signal view `v` is initialized and will try to use `z` as base and at the same time 16-byte strides because that's what was previously correct.

In summary: Signal views are defined with respect ot their original base, but the base array maybe copied into a different memory layout on initialization which requires things like strides to be adjusted.
